views.go

```diff
@@ -391,6 +397,70 @@ func (lk *logKeeper) viewAllLogs(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (lk *logKeeper) viewTestInDatabase(r *http.Request, buildID string, testID string) (*logFetchResponse, *apiError) {
```

Comment:
as an alternative to returning a pointer to the error we could return an `error` (which is an interface) and return an `apiError` in case of error. 
The caller could then cast/test its type and pull out the relevant fields.

Reply:
I was trying to be as restrictive as possible about the type returned and keep an interface of "the thing this returns will always be ready to be immediately serialized into a response". If the Go convention is to only return error I can switch to that but this keeps the calling code simpler.